components/Repo/Add.js

```diff
@@ -55,44 +82,144 @@ const styles = {
+      value: key,
+      text: t(`repo/add/template/${key}`, null, key)
+    }))
+  )
+  const [templateOptions, setTemplateOptions] = useState(schemaOptions)
+  const [templateFilter, setTemplateFilter] = useState('')
+  const [template, setTemplate] = useState({
+    value: schema,
+    text: t(`repo/add/template/${schema}`, null, schema)
+  })
+
+  useMemo(() => {
+    setTemplateOptions(
+      schemaOptions
+        .concat(
+          (data?.reposSearch?.nodes || []).map(repo => ({
+            value: repo.latestCommit.document.meta.template,
+            text:
+              repo.latestCommit.document.meta.title ||
+              repo.id.split('/')[1].replace('-', ' '),
+            repoId: repo.id,
+            slug: repo.latestCommit.document.meta.slug || repo.id.split('/')[1]
+          }))
+        )
+        .filter(
+          ({ text }) =>
+            !templateFilter ||
+            text.toLowerCase().includes(templateFilter.toLowerCase())
+        )
+    )
+  }, [data, templateFilter])
+
+  return isTemplate ? (
+    <Dropdown
+      label='Vorlage'
+      items={schemaOptions}
+      value={schema}
+      onChange={item => {
+        onChange({
+          schema: item.value,
+          templateRepoId: undefined,
+          templatePrefix: undefined
+        })
+      }}
+    />
+  ) : (
+    <Loader
+      error={data.error}
+      loading={data.loading}
+      render={() => (
+        <Autocomplete
+          label='Vorlage'
+          value={template}
+          filter={templateFilter}
+          items={templateOptions}
+          onChange={newTemplate => {
+            setTemplate(newTemplate)
+            setTemplateFilter('')
+            onChange({
+              schema: newTemplate.value,
+              templateRepoId: newTemplate.repoId,
+              templatePrefix: newTemplate.slug + '-'
+            })
+          }}
+          onFilterChange={newFilter => {
+            if (!template || template.text !== newFilter) {
+              setTemplateFilter(newFilter)
+            }
+          }}
+          icon={<SearchIcon size={30} style={{ color: colors.lightText }} />}
+        />
+      )}
+    />
+  )
+})
+
 class RepoAdd extends Component {
   constructor(props) {
     super(props)
+    const schema = schemaKeys.includes('article') ? 'article' : schemaKeys[0]
     this.state = {
-      title: '',
-      template: templateKeys.includes('article') ? 'article' : templateKeys[0]
+      schema,
+      title: ''
     }
   }
-  getSlug(title) {
-    const { template } = this.state
-    const schema = schemas[template]
-    const prefix = [REPO_PREFIX, schema && schema.repoPrefix]
-      .filter(Boolean)
-      .join('')
-    const slug = [prefix, slugify(title)].join('')
-
-    return slug
+  getSlug() {
+    const { title, schema, templatePrefix } = this.state
+    const { isTemplate } = this.props
+    const prefix = isTemplate
+      ? [REPO_PREFIX]
+      : [REPO_PREFIX, templatePrefix || schemas[schema]?.repoPrefix]
```

Comment:
Please don't hate me now 🙈 
```suggestion
    const prefix = (isTemplate
      ? [REPO_PREFIX, 'template-']
      : [REPO_PREFIX, templatePrefix || schemas[schema]?.repoPrefix])
```
(note the brackets to ensure that both are filtered and joined)

Would lead to:
```
template-binswanger
binswanger-jugend-ohne-gehorsam
```

What do you think?

Reply:
But then we need to process the templatePrefix to remove the `template-` as not to get `template-binswanger-jugend-phn-gehorsam` (since the slug of the template is now `template-binswanger`)